src/charm.py

```diff
@@ -462,6 +476,19 @@ def _bessd_config_file_content_matches(self, content: str) -> bool:
             return False
         return True
 
+    def _hwcksum_config_changed(self) -> bool:
+        try:
+            existing_content = json.loads(
+                self._bessd_container.pull(
+                    path=f"{BESSD_CONTAINER_CONFIG_PATH}/{CONFIG_FILE_NAME}"
+                ).read()
+            )
+        except PathError:
+            existing_content = {}
```

Comment:
I'd recommend the name to be closer to `_hwcksum_config_matches_pod`, not so much that it changed